src/ros/tf/republisher.cpp

```diff
@@ -35,7 +35,7 @@ TFRepublisher::~TFRepublisher()
 	thread_.join();
 	tick_thread_.join();
 	if(collection_) {
-		delete collection_;
+		collection_.reset();
```

Comment:
reset is not needed, just remove it everywhere. smart pointer deletes itself, no reset needed.